### ISSUE TITLE
Puppeteer Error

### ISSUE THREAD

**Issue by MxchuKing** [kylon/Sharedown#94]
Request Failed with status code 403

The video loads in Chromium, but it's different than before and it gives me this error while chromium closes

Sharedown, FFmpeg and YT-dlp are updated, but before updating it didn't work either 

Log attached
[sharedownLog.log](https://github.com/kylon/Sharedown/files/15204232/sharedownLog.log)

![Captura de pantalla 2024-05-03 150019](https://github.com/kylon/Sharedown/assets/168862821/6d6ac2f6-87e9-44c1-9662-d980d249e5a6)

**Comment by kylon** [comment]
are you getting the link from sharepoint context menu -> copy link?

**Comment by MxchuKing** [comment]
Yeah, followed the same steps that the previous times. Maybe something changed in the way the organization shares the video :c
I really tried everything, the video even starts to play, but after a couple of seconds gives me that error :c

**Comment by MxchuKing** [comment]
Yeah, followed the same steps that the previous times. Maybe something changed in the way the organization shares the video :c
I really tried everything, the video even starts to play, but after a couple of seconds gives me that error :c

**Comment by kylon** [comment]
i ll check later this week

**Comment by PeakeIT** [comment]
I'm getting the exact same error as well now. I'm trying to save a Microsoft Stream (recording of a Teams Meeting) which is stored in SharePoint.

**Comment by kylon** [comment]
so, log is clean
for some reason you are not allowed the fetch the video manifest and i don't know why

**Comment by tmsss** [comment]
I have a "Waiting falied" error from Puppeteer, I would post the log, but somehow this error doesn't generate any log file.

![imagem](https://github.com/user-attachments/assets/ee43fb0d-771d-41c8-acd4-3581f8dd30a7)

**Comment by kylon** [comment]
> I have a "Waiting falied" error from Puppeteer, I would post the log, but somehow this error doesn't generate any log file.
> 
> ![imagem](https://private-user-images.githubusercontent.com/735343/351732412-ee43fb0d-771d-41c8-acd4-3581f8dd30a7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE4MjkyNTQsIm5iZiI6MTcyMTgyODk1NCwicGF0aCI6Ii83MzUzNDMvMzUxNzMyNDEyLWVlNDNmYjBkLTc3MWQtNDFjOC1hY2Q0LTM1ODFmOGRkMzBhNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzI0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcyNFQxMzQ5MTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NzIxNWIwOTVlYzEwN2RkMjA1N2E2ZjMxY2E0MTkyYWMxMTVmMDNhOGQzZDZmZTgzZjQwOGViZWY1ZWZhYmJjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.ao7JViN_jFSGtAE2QrpSCxHX2N4b0I9u8gJ1eVwJPWY)

don't use automatic login, leave all fields empty and set basic login

**Comment by Nozavi** [comment]
@MxchuKing I see in your screenshot pasainvicto in the URL - are you in that organization? 
I'm asking because I'm getting the same error (and the log file is pretty much identical). What is particular about this video link (that's generating the error for me) is that it **belongs to a different organization than mine**. Because of that, the login process is also different, of course, which means that in order to login I had to enter my email and then enter **the code that was sent to my email address**.
Could any of this be causing the issue?

**Comment by tmsss** [comment]
> @MxchuKing I see in your screenshot pasainvicto in the URL - are you in that organization? I'm asking because I'm getting the same error (and the log file is pretty much identical). What is particular about this video link (that's generating the error for me) is that it **belongs to a different organization than mine**. Because of that, the login process is also different, of course, which means that in order to login I had to enter my email and then enter **the code that was sent to my email address**. @kylon Could any of this be causing the issue?

That seems to be the case for me, when I switch back to basic login as @kylon suggests, I get a Puppeteer "Request falied" error. 

Until there is fix for this, in case you want to manually download the videos, what worked for me was copying the videomanifest URL from the browser devtools and use ffmpeg in the command line like this (make sure that the videomanifest URL ends with "hybridPlayback=true)":

`ffmpeg -i "videomanifest-copied-from-devtools" -codec copy video.mp4`

**Comment by kylon** [comment]
@Nozavi login modules are not universal, thats why sharedown lets you add them, a different login page/process needs a specific login module for it

@tmsss can you post a log for the request error?

**Comment by tmsss** [comment]
The log reports an access denied error, but when I enable the automatic login I get a waited failed error as I reported above.

![imagem](https://github.com/user-attachments/assets/d0a769aa-d1c6-46cd-8285-c226744de6a4)

**Comment by kylon** [comment]
as i said login modules are not universal, your page may be different so a login for it must be written
are you getting the link from sharepoint context menu?

**Comment by ggbelotto** [comment]
> @MxchuKing I see in your screenshot pasainvicto in the URL - are you in that organization? I'm asking because I'm getting the same error (and the log file is pretty much identical). What is particular about this video link (that's generating the error for me) is that it **belongs to a different organization than mine**. Because of that, the login process is also different, of course, which means that in order to login I had to enter my email and then enter **the code that was sent to my email address**. @kylon Could any of this be causing the issue?

That seems to be the case. I'm also trying to download sharepoint videos from a different organization that asks me for a e-mail code. After i'm sucessfuly on the page, sharedown exhibits the puppeteer error 403.
